carbonmark-api/src/routes/projects/get.ts

```diff
@@ -88,26 +91,38 @@ const handler = (fastify: FastifyInstance) =>
     });
 
     /** Assign valid marketplace projects to map */
-    marketplaceProjectsData.projects.forEach((project) => {
-      if (
-        !isValidMarketplaceProject(project) ||
-        !CreditId.isValidProjectId(project.key)
-      ) {
-        return;
-      }
-      const [standard, registryProjectId] = project.key.split("-");
-      const { creditId: key } = new CreditId({
-        standard,
-        registryProjectId,
-        vintage: project.vintage,
+    marketplaceProjectsData.projects
+      /**
+       * Because marketplace projects do not contain country and category information we need to filter these
+       * to match only those for which there is an offset project that was filtered previously by country or category
+       * ... if country or category were provided in the query
+       */
+      .filter(({ id }) =>
+        args.category || args.country
+          ? // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- @todo Use CreditID
+            notNil(ProjectDataMap.get(id as CreditIdentifier))
+          : true
+      )
+      .forEach((project) => {
+        if (
+          !isValidMarketplaceProject(project) ||
+          !CreditId.isValidProjectId(project.key)
+        ) {
+          return;
+        }
+        const [standard, registryProjectId] = project.key.split("-");
```

Comment:
fyi the splitProjectId util already invokes isValidProjectId and throws so you might be able to shave that if/type guard a few lines up (not sure)
```suggestion
        const [standard, registryProjectId] = CreditId.splitProjectId(project.key);
```